### PR TITLE
Azure Firewall merged into network-september-release

### PR DESCRIPTION
**Pull request by fanymanea**
## Description

Adding Azure Firewall cmdlets (which were in preview until now), slightly modified for version 2018-08-01. This PR contains only the Application and Network Rule Collections. The new features available in 2018-08-01 will be added later.

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md [hunk 1]
```diff
@@ -64,6 +64,18 @@
 ## Version 6.3.1
 * Enable peering across Virtual Networks in multiple Tenants for Set/Add-AzureRmVirtualNetworkPeering
 
+## Version 6.4.0-preview
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ChangeLog.md]: can you remove the version header here and move these cmdlets under the [Current Release](https://github.com/Azure/azure-powershell/blob/2dca9c00f64f993da1bf296614259e06359a77aa/src/ResourceManager/Network/Commands.Network/ChangeLog.md#current-release) section

File: src/StackAdmin/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 2]
```diff
@@ -116,6 +116,12 @@
     <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.ApplicationGatewayTests\TestApplicationGatewayCRUD.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.AzureFirewallTests\TestAzureFirewallAllocateAndDeallocate.json">
```
Comment (praries880) [inline src/StackAdmin/Network/Commands.Network.Test/Commands.Network.Test.csproj]: undo all changes to stack, Stack team will pull in changes as required.

File: src/ResourceManager/Network/Stack/Commands.Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallAllocateAndDeallocate.json [hunk 3]
```diff
@@ -0,0 +1,7914 @@
+{
```
Comment (praries880) [inline src/ResourceManager/Network/Stack/Commands.Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallAllocateAndDeallocate.json]: undo this file

File: src/ResourceManager/Network/Stack/Commands.Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json [hunk 4]
```diff
@@ -0,0 +1,7655 @@
+{
```
Comment (praries880) [inline src/ResourceManager/Network/Stack/Commands.Network.Test/SessionRecords/Commands.Network.Test.ScenarioTests.AzureFirewallTests/TestAzureFirewallCRUD.json]: undo this file as well..

**Comment by praries880** [comment]
on demand : https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1088/

**Comment by praries880** [comment]
@MikhailTryakhov The changes look good to me... I have started the on demand job for the changes here. Do look at these and let me know if they look good to you as well.

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/AzureFirewallTests.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        public XunitTracingInterceptor _logger;
+
+        public AzureFirewallTests(ITestOutputHelper output)
+        {
+            _logger = new XunitTracingInterceptor(output);
+            XunitTracingInterceptor.AddToContext(new XunitTracingInterceptor(output));
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/AzureFirewallTests.cs]: can you add owner qualifiers for the firewall tests, example [here](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ApplicationGatewayTests.cs#L36)

**Comment by fanymanea** [comment]
https://azuresdkci.westus2.cloudapp.azure.com/view/az-ps-network-tests/job/ps-network-test-firewall/1/console

**Comment by praries880** [comment]
@MikhailTryakhov Kindly let us know when its good to merge the PR into the network September branch.

**Comment by fanymanea** [comment]
https://azuresdkci.westus2.cloudapp.azure.com/view/az-ps-network-tests/job/ps-network-test-firewall/2/

File: src/ResourceManager/Network/Commands.Network/AzureFirewall/ApplicationRule/NewAzureFirewallApplicationRuleCommand.cs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Commands.Network.Models;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmFirewallApplicationRule", SupportsShouldProcess = true), OutputType(typeof(PSAzureFirewallApplicationRule))]
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/AzureFirewall/ApplicationRule/NewAzureFirewallApplicationRuleCommand.cs]: Define the name like [this example](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/EndpointService/GetAzureRMAvailableEndpointServiceCommand.cs#L31). You'll need to do this for all cmdlet names.

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md [hunk 7]
```diff
@@ -22,6 +22,15 @@
 
 ## Version 6.7.0
 * Updated cmdlet New-AzureRmVirtualNetworkGatewayConnection with support for switch ExpressRouteGatewayBypass
+* Added new commands for feature: Azure Firewall via ARM
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ChangeLog.md]: move these under `Current Release`

**Comment by praries880** [comment]
@fanymanea  minor changes requested.

**Comment by praries880** [comment]
@fanymanea  You need to create alias mappings, use [this guide](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/create-alias-mappings.md)

**Comment by maddieclayton** [comment]
@fanymanea @praries880 Nope, that should fix the travis build.  @fanymanea Please contact me if you have any issues.

File: src/ResourceManager/Profile/Commands.Profile/AzureRmAlias/Mappings.cs [hunk 8]
```diff
@@ -760,6 +760,7 @@ public class Mappings
                        'Remove-AzNetworkSecurityGroup':  'Remove-AzureRmNetworkSecurityGroup',
                        'Get-AzLoadBalancerProbeConfig':  'Get-AzureRmLoadBalancerProbeConfig',
                        'New-AzVirtualNetworkGatewayConnection':  'New-AzureRmVirtualNetworkGatewayConnection',
+                       'New-AzFirewallNetworkRuleCollection':  'New-AzureRmFirewallNetworkRuleCollection',
```
Comment (praries880) [inline src/ResourceManager/Profile/Commands.Profile/AzureRmAlias/Mappings.cs]: @maddieclayton  can you validate the changes here?

**Comment by praries880** [comment]
@fanymanea  Kindly take care of the merge conflicts

**Pull request merged**